### PR TITLE
Adds a way for admins to keep the arrivals shuttle docked

### LINKED ISSUE #25211: Weird arrivals messages
(F) Leta Hastings (Assistant) has arrived at the station at Arrival Shuttle.
(F) [Common] Automated Announcement System coldly states, "Leta Hastings has signed up as Assistant"
(F) Kaden Philips (Assistant) has arrived at the station at Aft Starboard Maintenance.
(F) [Common] Automated Announcement System coldly states, "Kaden Philips has signed up as Assistant"
(F) Janette Linaria (Scientist) has arrived at the station at Research and Development.
(F) [Common] Automated Announcement System coldly states, "Janette Linaria has signed up as Scientist"
(F) Scuff Gerald (Assistant) has arrived at the station at Abandoned Library.
(F) [Common] Automated Announcement System coldly states, "Scuff Gerald has signed up as Assistant"
(F) Tonio Trendy (Cook) has arrived at the station at WARNING: PRESSURIZED DOORS.
(F) [Common] Automated

@Cyberboss

### PR DESCRIPTION
**Pull request by Cyberboss**
Fixes #25211

### CODE REVIEW

**Comment by RemieRichards** [comment]
put it on all shuttles.
because fug you.

**Comment by Cyberboss** [comment]
This PR isn't that general tho

**Comment by RemieRichards** [comment]
Yes, and I'm saying that shouldn't be the case.
For example, it could be used to lock the emergency shuttle inplace for BYOS rounds, without having to hack the timer to be 30 minutes, or any of the myriad other events admins may wish to lock shuttles for.

**Comment by Iamgoofball** [comment]
Literally why

You know this is going to be used to undo your feature right

**Comment by Cyberboss** [comment]
It's statlogged so if I see it being used too often, I'll just revert this.

**Comment by Iamgoofball** [comment]
They'll stickyban you like they did me

File: code/modules/shuttle/arrivals.dm [hunk 1]
```diff
@@ -173,4 +182,13 @@
...
 
 /obj/docking_port/mobile/arrivals/proc/QueueAnnounce(mob, rank)
-	LAZYADD(queued_announces, CALLBACK(GLOBAL_PROC, .proc/AnnounceArrival, mob, rank))
+	if(mode != SHUTTLE_CALL)
+		AnnounceArrival(mob, rank)
+	else
+		LAZYADD(queued_announces, CALLBACK(GLOBAL_PROC, .proc/AnnounceArrival, mob, rank))
+
+/obj/docking_port/mobile/arrivals/vv_edit_var(var_name, var_value)
+	if(var_name == "perma_docked")
```
Comment (lzimann) [inline code/modules/shuttle/arrivals.dm]: I don't think this is really needed, doubt anyone will be using VV to bypass feedback(and vv edits are logged and public anyway). Also, use to_chat() if you are going to keep this.

Comment (Iamgoofball) [inline code/modules/shuttle/arrivals.dm]: Absolutely keep this in JUST IN CASE

Comment (Iamgoofball) [inline code/modules/shuttle/arrivals.dm]: seriously i've had admins tell me they'll use VV to bypass checks before

Comment (lzimann) [inline code/modules/shuttle/arrivals.dm]: Oh, make this a switch aswell.

**Comment by optimumtact** [comment]
At this point you may as well just revert your original feature

**Pull request merged**